os/fs/smartfs/smartfs_utils.c

```diff
@@ -1125,13 +1151,13 @@ int smartfs_finddirentry(struct smartfs_mountpt_s *fs, struct smartfs_entry_s *d
  *
  ****************************************************************************/
 
-int smartfs_createentry(struct smartfs_mountpt_s *fs, uint16_t parentdirsector, struct smartfs_entry_s *direntry)
+int smartfs_createdirentry(struct smartfs_mountpt_s *fs, struct smartfs_entry_s *direntry)
 {
 	int ret;
 	uint16_t nextsector;
 
-	/* If we are in the context of smartfs_createentry, then smartfs_find_availableentry did not find an available entry.
-	   Createentry has to chain a new nextsector to parentdirsector, and allocate the first entry.
+	/* If we are in the context of smartfs_createdirentry, then smartfs_find_availableentry did not find an available entry.
+	   Createdirentry has to allocate a new nextsector for chaining to parentdirsector, and allocate the first entry.
```

Comment:
So here we can set 
```
direntry->last_dsector = direntry->dsector;
direntry->dsector = nextsector;
```
and when it sync or finish entry process,
```
if (direntry->last_dsector != direntry->dsector) {
    update_chain(direntry->last_dsector, direntry->dsector); 
}
```

Reply:
I think this is related to the first comment above?

> last_dsector is enough I think.
Actually dsector is parent entry, so instead of this, you'd better use dsector.

Or you  are referring to entry finding loop in find_availableentry?

Reply:
both !

Reply:
Ahh, ohkkay :P